tests/Unit/BucketTesting/StreamBucketLoggerTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\BucketTesting;
+
+use PHPUnit\Framework\TestCase;
+use WMDE\Fundraising\Frontend\BucketTesting\Bucket;
+use WMDE\Fundraising\Frontend\BucketTesting\Campaign;
+use WMDE\Fundraising\Frontend\BucketTesting\Logging\StreamBucketLogger;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FakeBucketLoggingEvent;
+
+/**
+ * @covers \WMDE\Fundraising\Frontend\BucketTesting\Logging\StreamBucketLogger
+ */
+class StreamBucketLoggerTest extends TestCase {
+
+	public function testLogWriterAddsDate() {
+		$logfile = fopen( 'php://memory', 'a+' );
```

Comment:
What about creating this in setUp? I did not check fully if that makes sense but noticed low level details in test method "arrange" part and repetition. Even if setUp is not suitable extracting this to avoid the low level details might be good.